homeassistant/components/samsungtv/config_flow.py

```diff
@@ -0,0 +1,184 @@
+                CONF_HOST: self._host,
+                CONF_ID: self._uuid,
+                CONF_IP_ADDRESS: self._ip,
+                CONF_MANUFACTURER: self._manufacturer,
+                CONF_METHOD: self._method,
+                CONF_MODEL: self._model,
+                CONF_NAME: self._name,
+                CONF_ON_ACTION: self._on_script,
+                CONF_PORT: self._port,
+            },
+        )
+
+    def _try_connect(self):
+        """Try to connect and check auth."""
+        for method in METHODS:
+            config = {
+                "name": "HomeAssistant",
+                "description": "HomeAssistant",
+                "id": "ha.component.samsung",
+                "host": self._host,
+                "method": method,
+                "port": self._port,
+                "timeout": 1,
+            }
+            try:
+                LOGGER.debug("Try config: %s", config)
+                with Remote(config.copy()):
+                    LOGGER.debug("Working config: %s", config)
+                    self._method = method
+                    return RESULT_SUCCESS
+            except AccessDenied:
+                LOGGER.debug("Working but denied config: %s", config)
+                return RESULT_AUTH_MISSING
+            except UnhandledResponse:
+                LOGGER.debug("Working but unsupported config: %s", config)
+                return RESULT_NOT_SUPPORTED
+            except (OSError):
+                LOGGER.debug("Failing config: %s", config)
+
+        LOGGER.debug("No working config found")
+        return RESULT_NOT_FOUND
+
+    async def async_step_import(self, user_input=None):
+        """Handle configuration by yaml file."""
+        self._on_script = user_input.get(CONF_ON_ACTION)
+        self._port = user_input.get(CONF_PORT)
+
+        return await self.async_step_user(user_input)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is not None:
+            ip_address = await self.hass.async_add_executor_job(
+                _get_ip, user_input[CONF_HOST]
+            )
+
+            await self.async_set_unique_id(ip_address)
+            self._abort_if_unique_id_configured()
+
+            self._host = user_input.get(CONF_HOST)
+            self._ip = self.context[CONF_IP_ADDRESS] = ip_address
+            self._title = user_input.get(CONF_NAME)
+
+            result = await self.hass.async_add_executor_job(self._try_connect)
+
+            if result != RESULT_SUCCESS:
+                return self.async_abort(reason=result)
+            return self._get_entry()
+
+        return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+    async def async_step_ssdp(self, user_input=None):
+        """Handle a flow initialized by discovery."""
+        host = urlparse(user_input[ATTR_SSDP_LOCATION]).hostname
+        ip_address = await self.hass.async_add_executor_job(_get_ip, host)
```

Comment:
How to load the configured values here if `async_step_import` failed because of TV was off.

Reply:
Please explain the scenario and problem in greater detail step by step.

Reply:
1. add `samsungtv` to yaml configuration
2. (re-)start HA with TV switched off
3. `async_step_import` is called but unable to connect to TV, so fails
4. TV is switched on
5. `async_step_ssdp` is getting called but doesn't know about the yaml configuration
6. device is setup but with the wrong config

Reply:
Doesn't the user have to confirm the ssdp discovery before that flow results in an entry? If the user doesn't want that, the user should not confirm and instead restart home assistant to let config yaml flow take precedence, right?

Reply:
This is what we know, but how should the user know about it?

Reply:
It should be clear from the message that the user confirms. We can tailor that message.

Reply:
Nevertheless the manual config was silently ignored.

Reply:
Silent how? Doesn't the user need to confirm the ssdp flow?

Reply:
Yes, you're right. Would be an update mechanism like for ssdp after import be helpful also vice versa? So you could "extent" the discovered device later with a `turn_on_action` or what else. Or are there plans to make all options available in the UI in the future?

Reply:
Updating an existing config entry from config yaml can be done via the config flow, if there's a unique id for each config entry. This works the same way as when we update from ssdp.

For GUI:
- We expose the options via the options flow, that will update the options.
- If fundamental config entry items need to change, the user needs to remove and set the entry up again via GUI. There's some discussion about this here:
https://github.com/home-assistant/architecture/issues/326

Reply:
OK